packages/libs/error-stack/src/fmt/hook.rs

```diff
@@ -681,15 +681,18 @@ fn into_boxed_hook<T: Send + Sync + 'static>(
                 .or_else(|| {
                     frame
                         .request_value::<T>()
-                        .as_ref()
-                        .map(|val| hook(val, ctx.cast()))
+                        .map(|ref val| hook(val, ctx.cast()))
                 })
         }
 
+        // emulate the behavior from nightly by searching for
+        //  - `Context::provide`: not available
+        //  - `Attachment`s: provide themself, emulated by `downcast_ref`
         #[cfg(not(nightly))]
-        {
-            frame.downcast_ref::<T>().map(|val| hook(val, ctx.cast()))
-        }
+        matches!(frame.kind(), crate::FrameKind::Attachment(_))
+            .then_some(frame)
+            .and_then(Frame::downcast_ref::<T>)
+            .map(|val| hook(val, ctx.cast()))
```

Comment:
This snippet will still work, correct? 

https://github.com/hashintel/hash/blob/75e4eb32eee27ddae237206360d80dc9ab15f4b8/packages/libs/error-stack/src/hook.rs#L264-L268

Is this something we want? imo yes.

Reply:
I designed this PR with https://github.com/hashintel/hash/pull/1103 in mind 🙂